src/pybind/mgr/cephadm/services/ingress.py

```diff
@@ -231,10 +234,34 @@ def keepalived_generate_config(
                     script = f'/usr/bin/curl {build_url(scheme="http", host=d.ip or "localhost", port=port)}/health'
         assert script
 
-        # set state. first host in placement is master all others backups
-        state = 'BACKUP'
-        if hosts[0] == host:
-            state = 'MASTER'
+        states = []
+        priorities = []
+        virtual_ips = []
+
+
+        # Set state and priority. Have one master for each VIP. Or at least the first one as master if only one VIP.
+        if spec.virtual_ip:
+            virtual_ips.append(spec.virtual_ip)
+            if hosts[0] == host:
+                states.append('MASTER')
+                priorities.append(100)
+            else:
+                states.append('BACKUP')
+                priorities.append(90)
+
+        else:
+            virtual_ips = spec.virtual_ips_list
```

Comment:
```
cephadm/services/ingress.py:253: error: Incompatible types in assignment (expression has type "Optional[List[str]]", variable has type "List[str]")
```
similar to other comment, need to make sure `spec.virtual_ips_list` is not `None` here for mypy.